Safetensor loader fails to compile due to API breaking change in `zip` crate

**Describe the bug**

Build fails when loading `safetensors` as described in https://burn.dev/books/burn/import/safetensors-model.html due to incompatibility with the new API from `zip`.

Minimal code to reproduce:

```
// build.rs

use burn::record::{FullPrecisionSettings, NamedMpkFileRecorder, Recorder};
use burn_import::safetensors::SafetensorsFileRecorder;

type Backend = burn_ndarray::NdArray<f32>;

fn main() {
    let device = Default::default();

    // Load from Safetensors
    let recorder = SafetensorsFileRecorder::<FullPrecisionSettings>::default();
    let record = recorder
        .load("some-safetensors-path".into(), &device)
        .expect("Should decode state successfully");

    // Save to Burn's binary format
    let recorder = NamedMpkFileRecorder::<FullPrecisionSettings>::default();
    recorder
        .record(record, "model.mpk".into())
        .expect("Failed to save model record");
}
```

The path to the source `safetensors` model is irrelevant for repro as this is a build failure:

`cargo build` fails with:

```sh
error[E0308]: mismatched types
  --> /mnt/data/local/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/burn-store-0.20.1/src/pytorch/lazy_data.rs:90:25
   |
87 |             file_list.push((name, offset, compressed_size));
   |             ---------      ------------------------------- this argument has type `(std::string::String, std::option::Option<u64>, u64)`...
   |             |
   |             ... which causes `file_list` to have type `std::vec::Vec<(std::string::String, std::option::Option<u64>, u64)>`
...
90 |         Ok(Self { path, file_list })
   |                         ^^^^^^^^^ expected `Vec<(String, u64, u64)>`, found `Vec<(String, Option<u64>, u64)>`
   |
   = note: expected struct `std::vec::Vec<(std::string::String, u64, _)>`
              found struct `std::vec::Vec<(std::string::String, std::option::Option<u64>, _)>`
```

**To Reproduce**
<!-- 
 Steps to reproduce the behavior:
 1. Go to '...'
 2. Click on '....'
 3. Scroll down to '....'
 4. See error
-->

**Expected behavior**
Build should succeed and port the `safetensors` model to `burn` format.

**Desktop (please complete the following information):**
 - OS: Linux
 - Browser N/A

```sh
(base) callpraths@personal-workstation:/mnt/data/src/nanogpt-rs$ uname -a
Linux personal-workstation 6.1.0-42-cloud-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.159-1 (2025-12-30) x86_64 GNU/Linux
```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -114,7 +114,7 @@ tracing-appender = "0.2.3"
 tracing-core = { version = "0.1.36", default-features = false }
 tracing-opentelemetry = "0.32.0"
 tracing-subscriber = "0.3.20"
-zip = "7.2.0"
+zip = "7.4.0"
 
 # Persist related
 memmap2 = { version = "0.9" }

